### PR TITLE
new custom dependency lookup for iconv

### PR DESCRIPTION
**Pull request by eli-schwartz**
Also internally needed by intl, so add that as a proxied dependency instead of coding it manually.

### CODE REVIEW

**Review by dcbaker — CHANGES_REQUESTED** [review]
This needs to be added to the lookup table, right?

**Comment by eli-schwartz** [comment]
Oh darn, knew I was forgetting something. :D

**Review by dcbaker — APPROVED** [review]

**Review by dcbaker — CHANGES_REQUESTED** [review]
Oh, wait, there's another docs page for built-in dependencies

**Review by dcbaker — APPROVED** [review]
Looks great. Thanks for bearing with me as i remember all of the things to do.

File: mesonbuild/dependencies/misc.py [hunk 1]
```diff
@@ -467,7 +486,9 @@ def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any]):
             self.is_found = True
 
             if self.static:
-                self.link_args += self.clib_compiler.find_library('iconv', env, [], self.libtype)
+                if not self._add_sub_dependency(iconv_factory(env, self.for_machine, {})):
```
Comment (lazka) [inline mesonbuild/dependencies/misc.py:489]: Could it be that this makes it no longer forward "static" to iconv?

Comment (dcbaker) [reply mesonbuild/dependencies/misc.py:489]: Yes, that would be the case

Comment (eli-schwartz) [reply mesonbuild/dependencies/misc.py:489]: I... think that was largely copy/paste from other uses of add_sub_dependency, which almost universally don't pass the libtype in.

Comment (dcbaker) [reply mesonbuild/dependencies/misc.py:489]: Likely, i think all the other uses are threads, which is kinda special

Comment (lazka) [reply mesonbuild/dependencies/misc.py:489]: -> #9722

**Pull request merged**